From ab1ffac8498084fccc7e6492c1d27d8aef92b17f Mon Sep 17 00:00:00 2001
From: Sivashankari Madhavan <quic_sivamadh@quicinc.com>
Date: Mon, 10 Oct 2022 09:36:36 +0530
Subject: [PATCH] ath12k: Fix avg ack info not shown in station dump

Ack signal station info is filling based on the extended feature
flag NL80211_EXT_FEATURE_ACK_SIGNAL_SUPPORT. It was not enabled
in ath12k. Due to this ack signal info, was skipping to print.

Fix it by enabling this flag in ath12k.

Station 00:03:7f:08:52:07 (on wlan2)
...
last ack signal:-95 dBm
last ack signal:-95 dBm
...

Signed-off-by: Sivashankari Madhavan <quic_sivamadh@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/mac.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/wireless/ath/ath12k/mac.c b/drivers/net/wireless/ath/ath12k/mac.c
index b494b62..1318792 100644
--- a/drivers/net/wireless/ath/ath12k/mac.c
+++ b/drivers/net/wireless/ath/ath12k/mac.c
@@ -11464,6 +11464,7 @@ static int ath12k_mac_hw_register(struct ath12k_hw *ah)
 
 	wiphy_ext_feature_set(hw->wiphy, NL80211_EXT_FEATURE_CQM_RSSI_LIST);
 	wiphy_ext_feature_set(hw->wiphy, NL80211_EXT_FEATURE_STA_TX_PWR);
+	wiphy_ext_feature_set(hw->wiphy, NL80211_EXT_FEATURE_ACK_SIGNAL_SUPPORT);
 
 	if (test_bit(WMI_TLV_SERVICE_BSS_COLOR_OFFLOAD, ab->wmi_ab.svc_map))
                 wiphy_ext_feature_set(hw->wiphy,
-- 
2.17.1

